pkg/destroy/kubevirt/destroyer.go

```diff
@@ -0,0 +1,136 @@
+package kubevirt
+
+import (
+	"github.com/sirupsen/logrus"
+
+	ickubevirt "github.com/openshift/installer/pkg/asset/installconfig/kubevirt"
+	"github.com/openshift/installer/pkg/destroy/providers"
+	"github.com/openshift/installer/pkg/types"
+)
+
+// ClusterUninstaller holds the various options for the cluster we want to delete.
+type ClusterUninstaller struct {
+	Metadata types.ClusterMetadata
+	Logger   logrus.FieldLogger
+}
+
+// Run is the entrypoint to start the uninstall process.
+func (uninstaller *ClusterUninstaller) Run() error {
+	namespace := uninstaller.Metadata.Kubevirt.Namespace
+	labels := uninstaller.Metadata.Kubevirt.Labels
+
+	kubevirtClient, err := ickubevirt.NewClient()
+	if err != nil {
+		return err
+	}
+	if err := uninstaller.deleteAllVMs(namespace, labels, kubevirtClient); err != nil {
+		return err
+	}
+	if err := uninstaller.deleteAllDVs(namespace, labels, kubevirtClient); err != nil {
+		return err
+	}
+	if err := uninstaller.deleteAllSecrets(namespace, labels, kubevirtClient); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (uninstaller *ClusterUninstaller) deleteAllVMs(namespace string, labels map[string]string, kubevirtClient ickubevirt.Client) error {
+	vmList, err := kubevirtClient.ListVirtualMachine(namespace)
+	if err != nil {
+		return err
+	}
+
+	var destroyList []string
+	for _, d := range vmList.Items {
+		existLabels := d.GetLabels()
+		for k, v := range labels {
+			if existVal, ok := existLabels[k]; ok && existVal == v {
+				destroyList = append(destroyList, d.GetName())
+				break
+			}
+		}
+	}
+
+	uninstaller.Logger.Infof("List tenant cluster's VMs (in namespace %s) return: %s", namespace, destroyList)
+	for _, vmName := range destroyList {
+		uninstaller.Logger.Infof("Delete VM %s", vmName)
+		if err := kubevirtClient.DeleteVirtualMachine(namespace, vmName); err != nil {
```

Comment:
k8s client support deleting with label selector, this will save you from all those loops:
```
    opts := &metav1.ListOptions{
		LabelSelector: labels.FormatLabels(map[string]string{"key", "val")),
	}
    // pass opts to dynamic client Delete function
```